daemon/rrd-update.c

```diff
@@ -243,34 +247,69 @@ void rb_rrd_update(rb_poller *poll)
                 log_errorx("couldn't create directory for raw file: %s : %s",
                             path,  strerror(errno));
                 free(parent);
-                /* Try the next item */
-                continue;
+                break; /* next raw file */
             }
             free(parent);
-        
-            fp = fopen(path, "a");
-            if(fp == NULL)
-            {
-                log_errorx("couldn't open raw file: %s for writing : %s",
-                            path,  strerror(errno));
-                /* Try the next item */
-                continue;
+
+            fd = open(path, O_WRONLY|O_APPEND|O_CREAT);
```

Comment:
According to open(2) documentation for O_CREAT: 

mode [...]  must  be  supplied  when O_CREAT or O_TMPFILE is specified in flags